sdk/go/x/auto/stack.go

```diff
@@ -0,0 +1,772 @@
+		res.Summary = history[0]
+	}
+
+	return res, nil
+}
+
+// Refresh compares the current stack’s resource state with the state known to exist in the actual
+// cloud provider. Any such changes are adopted into the current stack.
+func (s *Stack) Refresh(ctx context.Context, opts ...optrefresh.Option) (RefreshResult, error) {
+	var res RefreshResult
+
+	err := s.Workspace().SelectStack(ctx, s.Name())
+	if err != nil {
+		return res, errors.Wrap(err, "failed to refresh stack")
+	}
+
+	refreshOpts := &optrefresh.Options{}
+	for _, o := range opts {
+		o.ApplyOption(refreshOpts)
+	}
+
+	args := []string{"refresh", "--yes", "--skip-preview"}
+	if refreshOpts.Message != "" {
+		args = append(args, fmt.Sprintf("--message=%q", refreshOpts.Message))
+	}
+	for _, tURN := range refreshOpts.Target {
+		args = append(args, "--target %s", tURN)
+	}
+	if refreshOpts.Parallel > 0 {
+		args = append(args, fmt.Sprintf("--parallel=%d", refreshOpts.Parallel))
+	}
+
+	stdout, stderr, code, err := s.runPulumiCmdSync(ctx, args...)
+	if err != nil {
+		return res, newAutoError(errors.Wrap(err, "failed to refresh stack"), stdout, stderr, code)
+	}
+
+	history, err := s.History(ctx)
+	if err != nil {
+		return res, errors.Wrap(err, "failed to refresh stack")
+	}
+
+	var summary UpdateSummary
+	if len(history) > 0 {
+		summary = history[0]
+	}
+
+	res = RefreshResult{
+		Summary: summary,
+		StdOut:  stdout,
+		StdErr:  stderr,
+	}
+
+	return res, nil
+}
+
+// Destroy deletes all resources in a stack, leaving all history and configuration intact.
+func (s *Stack) Destroy(ctx context.Context, opts ...optdestroy.Option) (DestroyResult, error) {
+	var res DestroyResult
+
+	err := s.Workspace().SelectStack(ctx, s.Name())
+	if err != nil {
+		return res, errors.Wrap(err, "failed to destroy stack")
+	}
+
+	destroyOpts := &optdestroy.Options{}
+	for _, o := range opts {
+		o.ApplyOption(destroyOpts)
+	}
+
+	args := []string{"destroy", "--yes", "--skip-preview"}
+	if destroyOpts.Message != "" {
+		args = append(args, fmt.Sprintf("--message=%q", destroyOpts.Message))
+	}
+	for _, tURN := range destroyOpts.Target {
+		args = append(args, "--target %s", tURN)
+	}
+	if destroyOpts.TargetDependents {
+		args = append(args, "--target-dependents")
+	}
+	if destroyOpts.Parallel > 0 {
+		args = append(args, fmt.Sprintf("--parallel=%d", destroyOpts.Parallel))
+	}
+
+	stdout, stderr, code, err := s.runPulumiCmdSync(ctx, args...)
+	if err != nil {
+		return res, newAutoError(errors.Wrap(err, "failed to destroy stack"), stdout, stderr, code)
+	}
+
+	history, err := s.History(ctx)
+	if err != nil {
+		return res, errors.Wrap(err, "failed to destroy stack")
+	}
+
+	var summary UpdateSummary
+	if len(history) > 0 {
+		summary = history[0]
+	}
+
+	res = DestroyResult{
+		Summary: summary,
+		StdOut:  stdout,
+		StdErr:  stderr,
+	}
+
+	return res, nil
+}
+
+// Outputs get the current set of Stack outputs from the last Stack.Up().
+func (s *Stack) Outputs(ctx context.Context) (OutputMap, error) {
```

Comment:
Do you think it's necessary to drop the `Secret` bit? My thought here was to be as complete as possible (providing both the raw output value, and whether or not it's secret) and allow the user to remap or reconstruct the value if desired.

Reply:
Does the secretness impact the API consumer? If it's secret, should they not be able to get it, or at least provide some ack that they want to un-secret it?

Reply:
I assume if the user has privileges to perform the deployment, that they have privileges to read secrets that are a result of that deployment. If the API author wants to mask them from a downstream caller they can do so by mapping outputs into a custom struct.

Reply:
Thinking as a consumer of the outputs (human and API), what if autoapi split the outputs up into `outputs` (plaintext) and `secretOutputs` for the consumer?

This will still preserve the completeness of the output's properties mentioned, but raise the values up in the object's struct s.t. the value _is_ the outputs.

Reply:
Interestingly enough, the design you describe is what I started with initially. I got strong feedback from pat and luke to move towards an approach like this that combines the two. 

Given that we can easily see both sides of the fence here, I'd suggest maintaining this for the time being and seeing if we get any feedback from our community during this early alpha.